tests/analyser/analyser.cpp

```diff
@@ -0,0 +1,402 @@
+
+    EXPECT_EQ(libcellml::AnalyserModel::Type::INVALID, analyser->model()->type());
+}
+
+TEST(Analyser, variableInitialisedTwice)
+{
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("analyser/variable_initialised_twice.cellml"));
+
+    EXPECT_EQ(size_t(0), parser->issueCount());
+
+    const std::vector<std::string> expectedIssues = {
+        "Variable 'x' in component 'sub' and variable 'x' in component 'main' are equivalent and cannot therefore both be initialised.",
+    };
+    const std::vector<libcellml::Issue::Cause> expectedCauses = {
+        libcellml::Issue::Cause::VARIABLE,
+    };
+    const std::vector<libcellml::Issue::Level> expectedLevels = {
+        libcellml::Issue::Level::ERROR,
+    };
+
+    auto analyser = libcellml::Analyser::create();
+
+    analyser->analyseModel(model);
+
+    EXPECT_EQ_ISSUES_CAUSES_LEVELS(expectedIssues, expectedCauses, expectedLevels, analyser);
+
+    EXPECT_EQ(libcellml::AnalyserModel::Type::INVALID, analyser->model()->type());
+}
+
+TEST(Analyser, nonConstantInitialisingVariable)
+{
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("analyser/non_constant_initialising_variable.cellml"));
+
+    EXPECT_EQ(size_t(0), parser->issueCount());
+
+    const std::vector<std::string> expectedIssues = {
+        "Variable 'x' in component 'main' is initialised using variable 'k2', but it is not a constant.",
+    };
+    const std::vector<libcellml::Issue::Cause> expectedCauses = {
+        libcellml::Issue::Cause::VARIABLE,
+    };
+    const std::vector<libcellml::Issue::Level> expectedLevels = {
+        libcellml::Issue::Level::ERROR,
+    };
+
+    auto analyser = libcellml::Analyser::create();
+
+    analyser->analyseModel(model);
+
+    EXPECT_EQ_ISSUES_CAUSES_LEVELS(expectedIssues, expectedCauses, expectedLevels, analyser);
+
+    EXPECT_EQ(libcellml::AnalyserModel::Type::INVALID, analyser->model()->type());
+}
+
+TEST(Analyser, nonExistingInitialisingVariable)
+{
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("analyser/non_existing_initialising_variable.cellml"));
+
+    EXPECT_EQ(size_t(0), parser->issueCount());
+
+    const std::vector<std::string> expectedIssues = {
+        "Variable 'x' has an invalid initial value 'k'. Initial values must be a real number string or a variable reference.",
+    };
+    const std::vector<libcellml::Issue::Cause> expectedCauses = {
+        libcellml::Issue::Cause::VARIABLE,
+    };
+    const std::vector<libcellml::Issue::Level> expectedLevels = {
+        libcellml::Issue::Level::ERROR,
+    };
+
+    auto analyser = libcellml::Analyser::create();
+
+    analyser->analyseModel(model);
+
+    EXPECT_EQ_ISSUES_CAUSES_LEVELS(expectedIssues, expectedCauses, expectedLevels, analyser);
+
+    EXPECT_EQ(libcellml::AnalyserModel::Type::INVALID, analyser->model()->type());
+}
+
+TEST(Analyser, nonInitialisedState)
```

Comment:
In one of these tests it would be good to check if a variable is available from the issue and that its name is as expected.

Reply:
Done in commit c3f598ee.